elastic/kibana
# [Fleet] Fix package policy validation

## Summary

Part of #122679

Improve package policy validation to use variable definition from the package instead of the one provided by the user.
Improve the UI to no crash in case an invalidate package policy is here to allow user to fix it.

In a following PR I will add the package policy validation to the create and update package policy API where we do not validate the user provided info against the package.


## How to test

1. Create an invalid package policy (missing required variables)
```
curl --request POST \
  --url http://localhost:5601/api/fleet/package_policies \
  --header 'Authorization: Basic ZWxhc3RpYzpjaGFuZ2VtZQ==' \
  --header 'Content-Type: application/json' \
  --header 'kbn-xsrf: as' \
  --data '{
  "enabled": true,
  "inputs": [
    {
      "enabled": true,
      "streams": [
        {
          "data_stream": {
            "dataset": "auditd.log",
            "type": "logs"
          },
          "enabled": true,
          "vars": {
          }
        }
      ],
      "type": "logfile"
    }
  ],
  "name": "test-124",
  "namespace": "default",
  "output_id": "fleet-default-output",
  "package": {
    "name": "auditd",
    "title": "Auditd",
    "version": "1.3.0"
  },
  "policy_id": "a14e96b6-ad4c-5e47-aa64-6d3d8c8c6ef5"
}'
```

2. Visit the UI you should see validation errors and be able to fix them
<img width="943" alt="Screen Shot 2022-01-17 at 11 02 06 AM" src="https://user-images.githubusercontent.com/1336873/149803320-189c8113-0e73-44ca-805c-856a527c5df4.png">
